src/AmplienceImageStudio.ts

```diff
@@ -19,10 +29,31 @@ export type AmplienceImageStudioOptions = {
 };
 
 export class AmplienceImageStudio {
+  private eventListeners: Record<string, EventListenerCallback> = {};
```

Comment:
Should the key be an `ImageStudioEventType`?

```suggestion
  private eventListeners: Record<ImageStudioEventType, EventListenerCallback> = {};
```

Reply:
You cannot have `ImageStudioEventType` in here, otherwise it forces you to define, at compile type _all_ of the possible enum keys, which isn't what we want users to have to do . Event listeners are optional, and users should be able to specify the ones they want, maybe thats none atall.

Internally we also have some internal versions of the event handlers, that the user can override, the same applies here.

You will notice the EventHandlerCallbackConfig's however do specify `ImageStudioEventType` instead of `string`, this is intentional so that moving forwads we are intentional and explicit about the config - so this forces any devs to have to extend this list and add configs appropriately.

Good question tho 👍